test/functional/apps/discover/esql/_esql_view.ts

```diff
@@ -353,6 +353,9 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
           });
           await testSubjects.click('querySubmitButton');
           await header.waitUntilLoadingHasFinished();
+          // for some reason the chart query is taking a very long time to return (3x the delay)
+          // so wait for the chart to be loaded
+          await discover.waitForChartLoadingComplete(1);
```

Comment:
Why the new embeddable refactoring has created this delay? Does it mean that it affects the performance in ES|QL charts?

Reply:
Not that I can see in other scenarios or tests.
Only this specific test that manipulate the artificial `ELASTIC_ESQL_DELAY_SECONDS` variable seems to be affected somehow.

Reply:
I think this is the one that @lukasolson added.

@lukasolson or @jughosta does it seem right to you?

Reply:
Definitely seems odd. Is it always around 3x the delay even if you change the value for the delay?

Reply:
@dej611 have you possibly checked this? It would be good to know

Reply:
@dej611 Was the fetching logic changed for the chart? I think the same `ELASTIC_ESQL_DELAY_SECONDS` now affects the loading time of the chart data as it also calls `esql_async` with `error_query` to simulate the slowness:

<img width="1575" alt="Screenshot 2024-11-13 at 12 00 12" src="https://github.com/user-attachments/assets/8e005bc3-4bef-4538-b436-aca9d7439672">